### ISSUE TITLE
[XI]System.ComponentModel.DataAnnotations is crashing on a 32 bit device when llvm is enabled with release config

### ISSUE THREAD

**Issue by GouriKumari** [xamarin/xamarin-macios#3368]
### Steps to Reproduce

1. Execute System.ComponentModel.DataAnnotations on a 32 bit device with Release config and llvm enabled.

### Expected Behavior
Test should run successfully

### Actual Behavior
Test crashes on device

### Environment
Xamarin.iOS
Version: 11.8.0.18 (Visual Studio Community)
Hash: 4c56cbbf

Device: iPad2 9.3.5
This test passed on 64 bit devices iPhoneX  and iPadAir

### Build Logs
d15-6
Build Log: https://gist.github.com/GouriKumari/bdd81a93ac827bcc3196267100ad80bb
Crash Log: https://gist.github.com/GouriKumari/4eccfa6638e43a45ee148ec97ac7b2ce

##Regression Status: yes
d15-5
Build Log: https://gist.github.com/GouriKumari/496cad4f78b65d0060facf9c7a078d7e
Test Log: https://gist.github.com/GouriKumari/42f2ad692e4e572c32f0302b3cbefe80
Test ran successfully.

**Comment by dalexsoto** [comment]
I can reproduce what Gouri is seeing in an iPhone 5

Crash log: https://gist.github.com/dalexsoto/78a8f87e70a4bb85c078633e75e70b48#file-crash-log
Console log: https://gist.github.com/dalexsoto/78a8f87e70a4bb85c078633e75e70b48#file-console-output

/cc @spouliot FYI

**Comment by dalexsoto** [comment]
```
AssociationAttributeTest
 warning: mono_class_from_mono_type: implement me 0x00
 error: * Assertion: should not be reached at /Users/alex/monotouch/d156/xamarin-macios/external/mono/mono/metadata/class.c:6467
 critical: Stacktrace:
 critical:   at <unknown> <0xffffffff>
 critical:   at MonoTests.System.ComponentModel.DataAnnotations.AssociationAttributeTest.Constructor () <0x0080b>
 critical:   at (wrapper runtime-invoke) object.runtime_invoke_dynamic (intptr,intptr,intptr,intptr) <0x000e7>
 critical:   at <unknown> <0xffffffff>
 critical:   at (wrapper managed-to-native) System.Reflection.MonoMethod.InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&) <0x0004b>
 critical:   at System.Reflection.MonoMethod.Invoke (object,System.Reflection.BindingFlags,System.Reflection.Binder,object[],System.Globalization.CultureInfo) <0x000af>
 critical:   at System.Reflection.MethodBase.Invoke (object,object[]) <0x00027>
 critical:   at NUnit.Framework.Internal.Reflect.InvokeMethod (System.Reflection.MethodInfo,object,object[]) <0x0003f>
 critical:   at NUnit.Framework.Internal.Commands.TestMethodCommand.Execute (NUnit.Framework.Internal.TestExecutionContext) <0x00053>
 critical:   at NUnit.Framework.Internal.Commands.SetUpTearDownCommand.Execute (NUnit.Framework.Internal.TestExecutionContext) <0x00053>
 critical:   at NUnit.Framework.Internal.WorkItems.SimpleWorkItem.PerformWork () <0x00027>
 critical:   at NUnit.Framework.Internal.WorkItems.WorkItem.RunTest () <0x000fb>
 critical:   at NUnit.Framework.Internal.WorkItems.WorkItem.Execute (NUnit.Framework.Internal.TestExecutionContext) <0x0013b>
 critical:   at NUnit.Framework.Internal.WorkItems.CompositeWorkItem.RunChildren () <0x000f7>
 critical:   at NUnit.Framework.Internal.WorkItems.CompositeWorkItem.PerformWork () <0x003ab>
 critical:   at NUnit.Framework.Internal.WorkItems.WorkItem.RunTest () <0x000fb>
 critical:   at NUnit.Framework.Internal.WorkItems.WorkItem.Execute (NUnit.Framework.Internal.TestExecutionContext) <0x0013b>
 critical:   at NUnit.Framework.Internal.WorkItems.CompositeWorkItem.RunChildren () <0x000f7>
 critical:   at NUnit.Framework.Internal.WorkItems.CompositeWorkItem.PerformWork () <0x003ab>
 critical:   at NUnit.Framework.Internal.WorkItems.WorkItem.RunTest () <0x000fb>
 critical:   at NUnit.Framework.Internal.WorkItems.WorkItem.Execute (NUnit.Framework.Internal.TestExecutionContext) <0x0013b>
 critical:   at NUnit.Framework.Internal.WorkItems.CompositeWorkItem.RunChildren () <0x000f7>
 critical:   at NUnit.Framework.Internal.WorkItems.CompositeWorkItem.PerformWork () <0x003ab>
 critical:   at NUnit.Framework.Internal.WorkItems.WorkItem.RunTest () <0x000fb>
 critical:   at NUnit.Framework.Internal.WorkItems.WorkItem.Execute (NUnit.Framework.Internal.TestExecutionContext) <0x0013b>
 critical:   at MonoTouch.NUnit.UI.BaseTouchRunner.Run (NUnit.Framework.Internal.Test) <0x00123>
 critical:   at MonoTouch.NUnit.UI.BaseTouchRunner.Run () <0x00077>
 critical:   at MonoTouch.Dialog.StringElement.Selected (MonoTouch.Dialog.DialogViewController,UIKit.UITableView,Foundation.NSIndexPath) <0x00027>
 critical:   at MonoTouch.Dialog.DialogViewController.Selected (Foundation.NSIndexPath) <0x00093>
 critical:   at MonoTouch.Dialog.DialogViewController/Source.RowSelected (UIKit.UITableView,Foundation.NSIndexPath) <0x0003b>
 critical:   at (wrapper runtime-invoke) object.runtime_invoke_dynamic (intptr,intptr,intptr,intptr) <0x000e7>
 critical:   at <unknown> <0xffffffff>
 critical:   at (wrapper managed-to-native) UIKit.UIApplication.UIApplicationMain (int,string[],intptr,intptr) <0x0004f>
 critical:   at UIKit.UIApplication.Main (string[],string,string) <0x0009f>
 critical:   at BCL.Tests.Application.Main (string[]) <0x0001f>
 critical:   at (wrapper runtime-invoke) object.runtime_invoke_dynamic (intptr,intptr,intptr,intptr) <0x000e7>
```

**Comment by spouliot** [comment]
c.c. @vargaz

**Comment by vargaz** [comment]
Isn't this the same as:
https://github.com/xamarin/xamarin-macios/issues/3241

**Comment by dalexsoto** [comment]
@vargaz Could be but the crashlog is different

**Comment by rolfbjarne** [comment]
This test succeeds if I apply the fix for #3241 locally, so I'm closing as a dup of that one (for some reason the native stack trace here is actually present, which made it look like it was different from #3241).
